pandas/core/reshape/tile.py

```diff
@@ -444,7 +444,7 @@ def _coerce_to_type(x):
     if dtype is not None:
         # GH 19768: force NaT to NaN during integer conversion
         if is_bool_dtype(x):
```

Comment:
@jreback Not entirely sure where this should go - Adding `x.astype(int)` under  `elif is_bool_dtype(x)` higher up throws an error with the existing `np.where(x.notna(), x.view(np.int64), np.nan)` statement (its ok if x is Series but not an ndarray)
 `AttributeError: 'numpy.ndarray' object has no attribute 'isnan'`
    
It passes if is_bool_dtype(x) check with new `np.where` condition using `np.isnan(x)` to account for x being an ndarray.

Reply:
so don't use ``np.isnan``, change this to 

```x = np.where(notna(x), x.astype(np.int64, copy=False), np.nan)```

Reply:
I've gone with doing the integer conversion in the elif block higher up and leaving dtype as None as @jschendel suggested, rather than making any changes in the `dtype is not None` block.